src/lib/utils/redisCache.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-console */
+import { DataCache } from 'next-drupal'
+import { createClient, RedisClientType } from 'redis'
+
+export async function createRedisClient(url): Promise<RedisClientType> {
+  let isReady = false
+
+  let redisClient: RedisClientType
+  if (isReady === false) {
+    redisClient = createClient({ url })
+    redisClient.on('error', (err) => console.error(`Redis Error: ${err}`))
+    redisClient.on('connect', () => console.info('Redis connected'))
+    redisClient.on('reconnecting', () => console.info('Redis reconnecting'))
+    redisClient.on('ready', () => {
+      isReady = true
+      console.info('Redis ready!')
+    })
+  }
+
+  await redisClient.connect()
+
+  return redisClient
+}
+
+export function redisCache(client: Promise<RedisClientType>): DataCache {
+  let redis: RedisClientType
+
+  client
+    .then((connection) => {
+      redis = connection
+    })
+    .catch((err) => {
+      console.error({ err }, 'Failed to connect to Redis')
+    })
+
+  return {
+    async set(key, value) {
+      return await (await client).set(key, value)
```

Comment:
Do we need to `await` the `.set` call? Since this is `await`ed in the next-drupal code, I think we can just return the promise rather than wait for a resolved promise. It ultimately doesn't matter, but it might make things a bit cleaner here than having a double `await`.